examples/MagicWeather/src/screens/WeatherScreen.tsx

```diff
@@ -38,35 +31,65 @@ const WeatherScreen = () => {
       if (typeof customerInfo.entitlements.active[ENTITLEMENT_ID] !== 'undefined') {
         setWeatherData(generateSampleData(Environment.EARTH));
       } else {
-        navigation.navigate('Paywall');
+        navigation.openModal('Paywall');
```

Comment:
Did you test this and it works? because we've been dealing with tons of issues with the Paywall component 

https://github.com/RevenueCat/react-native-purchases/issues/1162